pkg/virt-controller/watch/vmi.go

```diff
@@ -2447,3 +2449,51 @@ func (c *VMIController) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 		vmi.Labels[virtv1.MemoryHotplugOverheadRatioLabel] = *overheadRatio
 	}
 }
+
+func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
+	if len(dvs) == 0 {
+		return
+	}
+
+	dvsReadyCondition := virtv1.VirtualMachineInstanceCondition{
+		Status:  k8sv1.ConditionTrue,
+		Type:    virtv1.VirtualMachineInstanceDataVolumesReady,
+		Reason:  virtv1.VirtualMachineInstanceReasonAllDVsReady,
+		Message: "All of the VMI's DVs are bound and not running",
+	}
+
+	var dvMessages []string
+	for _, dv := range dvs {
+		cStatus, cMessage := statusAndMessageOfReadyCondition(dv.Status.Conditions)
+		if cStatus != k8sv1.ConditionTrue {
+			dvsReadyCondition.Reason = virtv1.VirtualMachineInstanceReasonNotAllDVsReady
+			if cMessage != "" {
+				dvMessages = append(dvMessages, cMessage)
+			}
+			if cStatus == k8sv1.ConditionFalse {
+				dvsReadyCondition.Status = cStatus
+			} else if cStatus == k8sv1.ConditionUnknown && dvsReadyCondition.Status == k8sv1.ConditionTrue {
+				dvsReadyCondition.Status = cStatus
+			}
+		}
+	}
+
+	if dvsReadyCondition.Status != k8sv1.ConditionTrue {
+		dvsReadyCondition.Message = "Not all of the VMI's DVs are ready"
+		if len(dvMessages) > 0 {
+			dvsReadyCondition.Message += ": " + strings.Join(dvMessages, ",")
```

Comment:
I am a little bit worried about the readability of this.

Reply:
OK, let's remove it to always have a predictable message.